libs/wingsdk/src/target-sim/queue.inflight.ts

```diff
@@ -35,9 +35,11 @@ export class Queue
   }
 
   public async cleanup(): Promise<void> {
-    clearInterval(this.intervalId);
+    await this.processLoop.stop();
   }
```

Comment:
I thought there might be some flakiness caused by the current polling implementation, because when `cleanup()` calls `clearInterval()`, it doesn't wait for the current `processMessages` call to finish, meaning the simulator could end up in a limbo state. To that end I tried adding a hand-written poller that let the `cleanup()` function wait for the loop to finish before the simulator continues cleaning up other resources.

Reply:
Draining :-)

Reply:
It was a fun function to write with the help of ChatGPT 😄